enterprise/frontend/src/metabase-enterprise/documents/components/Editor/extensions/Command/CommandSuggestion.tsx

```diff
@@ -0,0 +1,550 @@
+      } else {
+        if (index < commandOptions.length) {
+          executeCommand(commandOptions[index].command);
+        }
+      }
+    }
+  };
+
+  const upHandler = () => {
+    setSelectedIndex((prev) => (prev - 1 + totalItems) % totalItems);
+  };
+
+  const downHandler = () => {
+    setSelectedIndex((prev) => (prev + 1) % totalItems);
+  };
+
+  const enterHandler = () => {
+    selectItem(selectedIndex);
+  };
+
+  useEffect(() => {
+    setSelectedIndex(0);
+  }, [
+    currentItems.length,
+    showLinkSearch,
+    showEmbedSearch,
+    searchMenuItems.length,
+  ]);
+
+  useEffect(() => {
+    const selectedItem = itemRefs.current[selectedIndex];
+    if (selectedItem) {
+      selectedItem.scrollIntoView({ block: "nearest", behavior: "smooth" });
+    }
+  }, [selectedIndex]);
+
+  useImperativeHandle(ref, () => ({
+    onKeyDown: ({ event }: { event: KeyboardEvent }) => {
+      if (showLinkSearch || showEmbedSearch) {
+        return entityHandlers.onKeyDown({ event });
+      }
+
+      if (event.key === "ArrowUp") {
+        upHandler();
+        return true;
+      }
+
+      if (event.key === "ArrowDown") {
+        downHandler();
+        return true;
+      }
+
+      if (event.key === "Enter") {
+        enterHandler();
+        return true;
+      }
+
+      return false;
+    },
+  }));
+
+  if ((showLinkSearch || showEmbedSearch) && isLinkSearchLoading) {
+    return <LoadingSuggestionPaper aria-label={t`Command Dialog`} />;
+  }
+
+  return (
+    <SuggestionPaper aria-label={t`Command Dialog`}>
+      {showLinkSearch || showEmbedSearch ? (
+        <EntitySearchSection
+          menuItems={linkMenuItems}
+          selectedIndex={entitySelectedIndex}
+          onItemSelect={entityHandlers.selectItem}
+          onFooterClick={entityHandlers.openModal}
+          query={effectiveQuery}
+          searchResults={searchResults}
+          modal={entityModal}
+          onModalSelect={entityHandlers.handleModalSelect}
+          onModalClose={entityHandlers.handleModalClose}
+        />
+      ) : (
+        <>
+          {commandOptions.length > 0 ||
+          (query && searchMenuItems.length > 0) ? (
+            <>
+              {query && searchMenuItems.length > 0 ? (
+                // When searching, show question search results first, then matching commands
+                <>
+                  {searchMenuItems.map((item, index) => (
+                    <MenuItemComponent
+                      key={`search-${index}`}
+                      item={item}
+                      isSelected={selectedIndex === index}
+                      onClick={() => selectItem(index)}
+                    />
+                  ))}
+                  {searchMenuItems.length > 0 && commandOptions.length > 0 && (
+                    <Divider my="sm" mx="sm" />
+                  )}
+                  {commandOptions.map((option, cmdIndex) => {
+                    const index = searchMenuItems.length + cmdIndex;
```

Comment:
Menu is not properly abstracted, we hardcode sections, their order and then compute index in the entire list of available items like this